Desktop/analysis/analysis.cpp

```diff
@@ -431,59 +434,66 @@ Json::Value& Analysis::_getParentBoundValue(const QVector<JASPControl::ParentKey
 						row[parent.key] = newValue;
 					}
 					parentBoundValues.append(row);
-					result = &(parentBoundValues[parentBoundValues.size() - 1]);
+					parentBoundValue = &(parentBoundValues[parentBoundValues.size() - 1]);
 					found = true;
 				}
 			}
 		}
 	}
 
-	return *result;
+	return result;
 }
 
 bool Analysis::setBoundValue(const std::string &name, const Json::Value &value, const Json::Value &meta, const QVector<JASPControl::ParentKey>& parentKeys)
 {
 	bool found = false;
-	Json::Value &	parentBoundValue	= _getParentBoundValue(parentKeys, found, true),
-					copyPBV				= parentBoundValue; 
+	Json::Value* parentBoundValue	= &_boundValues;
+	QVector<std::string> parents	= _getParentBoundValue(parentKeys, parentBoundValue, found, true);
```

Comment:
Isnt it actually more sensible to give the child-values to the parent and let the parent add them?

Instead of letting the child get a weird ref to a Json::Value in its parent and then setting stuff on there?